spacetemplate/repository.go

```diff
@@ -0,0 +1,286 @@
+	// it
+	parsedSpaceTempl, err := spaceTempl.Parse()
+	if err != nil {
+		log.Info(ctx, map[string]interface{}{
+			"space_template_id": spaceTemplateID,
+			"err":               err,
+		}, "failed to parse space template")
+		return errors.NewBadParameterError("space_template", spaceTempl.Template).Expected("parse-able space template")
+	}
+	if err = parsedSpaceTempl.GenerateNewIDs(); err != nil {
+		log.Info(ctx, map[string]interface{}{
+			"space_template_id": spaceTemplateID,
+			"err":               err,
+		}, "failed to generate IDs for space template")
+		return errors.NewInternalError(err.Error())
+	}
+	//fmt.Printf("with IDs:\n%s", parsedSpaceTempl)
+	if err = parsedSpaceTempl.Validate(); err != nil {
+		log.Info(ctx, map[string]interface{}{
+			"space_template_id": spaceTemplateID,
+			"err":               err,
+		}, "parsed space template not valid")
+		return errors.NewBadParameterError("parsed_space_template", parsedSpaceTempl.String()).Expected("valid parsed space template")
+	}
+
+	// Create WITs
+	witRepo := workitem.NewWorkItemTypeRepository(r.db)
+	for _, witHelper := range parsedSpaceTempl.WITs {
+		wit := witHelper.WorkItemType
+		var extendsTypePtr *uuid.UUID
+		if witHelper.ExtendsType != "" {
+			et, err := uuid.FromString(witHelper.ExtendsType)
+			if err != nil {
+				log.Info(ctx, map[string]interface{}{
+					"space_template_id": spaceTemplateID,
+					"wit_name":          wit.Name,
+					"space_id":          spaceID,
+					"err":               err,
+				}, "failed to parse work item type ID from string")
+				return errs.Wrapf(err, "failed to create work item type '%s' from space template '%s' in space '%s'", wit.Name, spaceTemplateID, spaceID)
+			}
+			extendsTypePtr = &et
+		}
+
+		witCreated, err := witRepo.Create(ctx, spaceID, &wit.ID, extendsTypePtr, wit.Name, wit.Description, wit.Icon, wit.Fields)
+		if err != nil {
+			log.Info(ctx, map[string]interface{}{
+				"space_template_id": spaceTemplateID,
+				"wit_name":          wit.Name,
+				"space_id":          spaceID,
+				"err":               err,
+			}, "failed to create work item type")
+			return errs.Wrapf(err, "failed to create work item type '%s' from space template '%s' in space '%s'", wit.Name, spaceTemplateID, spaceID)
+		}
+		if deleteArtifacts {
+			witToDelete := workitem.WorkItemType{ID: witCreated.ID}
+			defer r.db.Unscoped().Delete(witToDelete)
+		}
+	}
+
+	// Create WILTs
+	// Enfore all of them to be in the "user" link category
+	var linkCat *link.WorkItemLinkCategory
+	if len(parsedSpaceTempl.WILTs) > 0 {
+		linkCat, err = link.NewWorkItemLinkCategoryRepository(r.db).LoadCategoryFromDB(ctx, link.SystemWorkItemLinkCategoryUser)
+		if err != nil {
+			return errs.WithStack(err)
+		}
+	}
+	wiltRepo := link.NewWorkItemLinkTypeRepository(r.db)
+	for _, wiltHelper := range parsedSpaceTempl.WILTs {
+		if linkCat == nil {
+			return errs.Errorf("failed to dind work item link category: %s", link.SystemWorkItemLinkCategoryUser)
```

Comment:
There's an error but it should be `find` and not `bind` or `dind`. Fixed in bad54efb1c148424de113f7a1638f88a1d2ac8cb